spyder/utils/encoding.py

```diff
@@ -235,10 +235,25 @@ def write(text, filename, encoding='utf-8', mode='wb'):
         with open(filename, mode) as textfile:
             textfile.write(text)
     else:
+        # Based in the solution at untitaker/python-atomicwrites#42
+        # Needed to fix file permissions overwritting
+        # See spyder-ide/spyder#9381
+        try:
+            FileNotFoundError
+        except NameError:  # Python 2
+            FileNotFoundError = OSError

```

Comment:
@dalthviz, this is very convoluted. Let's use `OSError` for now and let's leave a comment below saying that we should really use `FileNotFoundError` when we move to Python 3.